processor/tailsamplingprocessor/composite_helper_test.go

```diff
@@ -16,86 +16,71 @@ package tailsamplingprocessor
-	"go.opentelemetry.io/collector/config"
 	"go.uber.org/zap"
 )
 
 func TestCompositeHelper(t *testing.T) {
-	cfg := &Config{
-		ProcessorSettings:       config.NewProcessorSettings(config.NewComponentID(typeStr)),
-		DecisionWait:            10 * time.Second,
-		NumTraces:               100,
-		ExpectedNewTracesPerSec: 10,
-		PolicyCfgs: []PolicyCfg{
-			{
-				Name: "composite-policy-1",
-				Type: Composite,
-				CompositeCfg: CompositeCfg{
-					MaxTotalSpansPerSecond: 1000,
-					PolicyOrder:            []string{"test-composite-policy-1", "test-composite-policy-2", "test-composite-policy-3", "test-composite-policy-4", "test-composite-policy-5", "test-composite-policy-6", "test-composite-policy-7", "test-composite-policy-8"},
-					SubPolicyCfg: []SubPolicyCfg{
-						{
-							Name:                "test-composite-policy-1",
-							Type:                NumericAttribute,
-							NumericAttributeCfg: NumericAttributeCfg{Key: "key1", MinValue: 50, MaxValue: 100},
-						},
-						{
-							Name:               "test-composite-policy-2",
-							Type:               StringAttribute,
-							StringAttributeCfg: StringAttributeCfg{Key: "key2", Values: []string{"value1", "value2"}},
-						},
-						{
-							Name:            "test-composite-policy-3",
-							Type:            RateLimiting,
-							RateLimitingCfg: RateLimitingCfg{SpansPerSecond: 10},
-						},
-						{
-							Name:             "test-composite-policy-4",
-							Type:             Probabilistic,
-							ProbabilisticCfg: ProbabilisticCfg{HashSalt: "salt", SamplingPercentage: 10},
-						},
-						{
-							Name:       "test-composite-policy-5",
-							Type:       Latency,
-							LatencyCfg: LatencyCfg{ThresholdMs: 10},
-						},
-						{
-							Name:          "test-composite-policy-6",
-							Type:          StatusCode,
-							StatusCodeCfg: StatusCodeCfg{StatusCodes: []string{"200", "404"}},
-						},
-						{
-							Name: "test-composite-policy-7",
-							Type: AlwaysSample,
-						},
-						{
-							Name: "test-composite-policy-8",
-						},
+	t.Run("valid", func(t *testing.T) {
+		actual, err := getNewCompositePolicy(zap.NewNop(), &CompositeCfg{
+			MaxTotalSpansPerSecond: 1000,
+			PolicyOrder:            []string{"test-composite-policy-1"},
+			SubPolicyCfg: []CompositeSubPolicyCfg{
+				{
+					sharedPolicyCfg: sharedPolicyCfg{
+						Name:       "test-composite-policy-1",
+						Type:       Latency,
+						LatencyCfg: LatencyCfg{ThresholdMs: 100},
+					},
+				},
+				{
+					sharedPolicyCfg: sharedPolicyCfg{
+						Name:       "test-composite-policy-2",
+						Type:       Latency,
+						LatencyCfg: LatencyCfg{ThresholdMs: 200},
 					},
-					RateAllocation: []RateAllocationCfg{
-						{
-							Policy:  "test-composite-policy-1",
-							Percent: 50,
-						},
-						{
-							Policy:  "test-composite-policy-2",
-							Percent: 25,
-						},
+				},
+			},
+			RateAllocation: []RateAllocationCfg{
+				{
+					Policy:  "test-composite-policy-1",
```

Comment:
Is this related to the other policy-1? Or should it be policy-3?

Reply:
I am afraid I don't understand the question. The name is equal to the name of the policy so that the rates can be properly allocated because the implementation looks up the values based on their name.